### PR TITLE
Test: Update Vagrant box to version 83

### LINKED ISSUE #4429: CI: CNI plugin test with version 83 of CI vagrant VM fails
Reproducing steps:

1. checkout latest master (commit `1a9be941a03d80fad43d9d6cd5aff234ba260aa9`
2. Run the following command: `cd test; vagrant up runtime`
3. Run the following command: `ginkgo --focus="RuntimeValidatedConnectivityTest" --focus="With CNI" -v -- -cilium.holdEnvironment=true -cilium.provision=false`

The test will fail:

```
level=info msg="environment variable \"K8S_VERSION\" was not set; setting to default value \"1.10\""
Running Suite: runtime
======================
Random Seed: 1528509170
Will run 1 of 288 specs

SSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSS
------------------------------
RuntimeValidatedConnectivityTest With CNI 
  Basic connectivity test
  /Users/ianvernon/go/src/github.com/cilium/cilium/test/ginkgo-ext/scopes.go:373
STEP: Importing policy
STEP: Setting up policy: /home/vagrant/go/src/github.com/cilium/cilium/test/CNI-policy.json
STEP: Adding containers

---
RuntimeValidatedConnectivityTest With CNI Basic connectivity test
at /Users/ianvernon/go/src/github.com/cilium/cilium/test/ginkgo-ext/scopes.go:373

CNI exec-plugins did not work correctly
Expected command: sudo -E PATH=$PATH:/opt/cni/bin -E CNI_PATH=/tmp/tmp.YpfbtsAzax/bin /tmp/tmp.YpfbtsAzax/cni/scripts/exec-plugins.sh add 5be28fb0630e8d7eeaea350a444dba57f296b1d45ef09e5da11f224c0ad3e264 /proc/11361/ns/net 
To succeed, but it failed:
Exitcode: 1 
Stdout:
 	 crio-bridge : error executing ADD: 
	 
Stderr:
 	 Usage: bridge [ OPTIONS ] OBJECT { COMMAND | help }
	        bridge [ -force ] -batch filename
	 where	OBJECT := { link | fdb | mdb | vlan | monitor }
	 	OPTIONS := { -V[ersion] | -s[tatistics] | -d[etails] |
	 		     -o[neline] | -t[imestamp] | -n[etns] name |
	 		     -c[ompressvlans] -j{son} }
	 


Pausing test for debug, use vagrant to access test setup.
Run "kill -SIGCONT 9667" to continue.
```

I looked at the script being executed: 
```
/tmp/tmp.YpfbtsAzax/cni/scripts/exec-plugins.sh 
```

I opened the file:
```
#!/usr/bin/env bash

if [[ ${DEBUG} -gt 0 ]]; then set -x; fi

NETCONFPATH=${NETCONFPATH-/etc/cni/net.d}

function exec_plugins() {
        i=0
        contid=$2
        netns=$3
        export CNI_COMMAND=$(echo $1 | tr '[:lower:]' '[:upper:]')
        export PATH=$CNI_PATH:$PATH
        export CNI_CONTAINERID=$contid
        export CNI_NETNS=$netns

        for netconf in $(echo $NETCONFPATH/*.conf | sort); do
                name=$(jq -r '.name' <$netconf)
                plugin=$(jq -r '.type' <$netconf)
                export CNI_IFNAME=$(printf eth%d $i)

                res=$($plugin <$netconf)
                if [ $? -ne 0 ]; then
                        errmsg=$(echo $res | jq -r '.msg')
                        if [ -z "$errmsg" ]; then
                                errmsg=$res
                        fi

                        echo "${name} : error executing $CNI_COMMAND: $errmsg"
                        exit 1
                elif [[ ${DEBUG} -gt 0 ]]; then
                        echo ${res} | jq -r .
                fi

                let "i=i+1"
        done
}

if [ $# -ne 3 ]; then
        echo "Usage: $0 add|del CONTAINER-ID NETNS-PATH"
        echo "  Adds or deletes the container specified by NETNS-PATH to the networks"
        echo "  specified in \$NETCONFPATH directory"
        exit 1
fi

exec_plugins $1 $2 $3
```

The error string which shows in the test failure is from:

```
echo "${name} : error executing $CNI_COMMAND: $errmsg"
```

Which means that the command that failed was:

```
res=$($plugin <$netconf)
```

This suggests that the cri-o installation in the VM starting in VM version 78 (which was built here: https://jenkins.cilium.io/job/Vagrant-Master-Boxes-Packer-Build/78/) added this file, which is causing the test to fail:
```
$ cat /etc/cni/net.d/100-crio-bridge.conf 
{
    "cniVersion": "0.3.0",
    "name": "crio-bridge",
    "type": "bridge",
    "bridge": "cni0",
    "isGateway": true,
    "ipMasq": true,
    "ipam": {
        "type": "host-local",
        "subnet": "10.88.0.0/16",
        "routes": [
            { "dst": "0.0.0.0/0" }
        ]
    }
}

```

Fix the image / provisioning of the VM in the Cilium repo to account for the presence of this file which is causing the CNI test to fail.

### PR DESCRIPTION
**Pull request by eloycoto**
Updated Vagrant box to version 83, and fix the issue with CNI.

Fix #4429

Signed-off-by: Eloy Coto <eloy.coto@gmail.com>

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/4432)

### CODE REVIEW

**Comment by eloycoto** [comment]
@rlenglet feel free to cherry-pick my commit if you need to fix another issue

**Comment by eloycoto** [comment]
test-me-please

**Review by aanm — APPROVED** [review]

**Comment by eloycoto** [comment]
test-me-please

**Comment by eloycoto** [comment]
https://jenkins.cilium.io/job/Cilium-PR-Ginkgo-Tests-Validated/3974/

**Comment by eloycoto** [comment]
https://jenkins.cilium.io/job/Cilium-PR-Ginkgo-Tests-Validated/3975/testReport/k8s-1/7/K8sValidatedServicesTest_Bookinfo_Demo_Tests_bookinfo_demo/

**Comment by eloycoto** [comment]
test-me-please

**Comment by eloycoto** [comment]
https://jenkins.cilium.io/job/Cilium-PR-Ginkgo-Tests-Validated/3976/testReport/

**Comment by eloycoto** [comment]
test-me-please

**Comment by manalibhutiyani** [comment]
Hit https://jenkins.cilium.io/job/Cilium-PR-Ginkgo-Tests-Validated/3981/testReport/k8s-1/7/K8sValidatedTunnelTest_VXLan_Check_VXLAN_mode/

**Comment by manalibhutiyani** [comment]
test-me-please

**Comment by eloycoto** [comment]
test-me-please

File: test/runtime/connectivity.go [hunk 1]
```diff
@@ -188,8 +188,23 @@ var _ = Describe("RuntimeValidatedConnectivityTest", func() {
...
 			cniClient   = "cni-client"
+			netDPath    = "/etc/cni/net.d/"
+			tmpDir      *helpers.CmdRes
 		)
-		var tmpDir *helpers.CmdRes
+
+		BeforeAll(func() {
+			// Remove any CNI plugin installed in the provision server. This
+			// helps to avoid issues on installing the new CNI
+			_ = vm.ExecWithSudo(fmt.Sprintf("rm -rf %[1]s/*.conf", netDPath)).ExpectSuccess(
```
Comment (ianvernon) [inline test/runtime/connectivity.go:198]: Ideally it shouldn't, but does this affect cri-o support within the VM after the test is ran? Furthermore, if users run this locally (which is a supported option as specified in the Ginkgo contributing guides), we may be removing configuration files that users have installed locally.

**Comment by eloycoto** [comment]
test-me-please

**Comment by rlenglet** [comment]
You can’t just update just the Vagrant box without also updating the Envoy SHA. Both must be modified together. Otherwise you’ll get timeouts because the Bazel cache is not in sync.

**Comment by ianvernon** [comment]
@rlenglet where can we find out how to update the Envoy SHA?

**Comment by rlenglet** [comment]
Just revert https://github.com/cilium/cilium/pull/4430 as a whole, not just parts of it.

**Comment by eloycoto** [comment]
@ianvernon @rlenglet should be ok now ;-)

**Comment by eloycoto** [comment]
test-me-please

**Review by rlenglet — APPROVED** [review]
LGTM, thanks!

**Comment by eloycoto** [comment]
https://jenkins.cilium.io/job/Cilium-PR-Ginkgo-Tests-Validated/4050/

**Comment by eloycoto** [comment]
test-me-please

**Comment by rlenglet** [comment]
CI failed build:
https://jenkins.cilium.io/job/Cilium-PR-Ginkgo-Tests-Validated/4056/
```
19:29:14 K8sValidatedTunnelTest
19:29:14 /home/jenkins/workspace/Cilium-PR-Ginkgo-Tests-Validated/src/github.com/cilium/cilium/test/ginkgo-ext/scopes.go:341
19:29:14   Geneve
19:29:14   /home/jenkins/workspace/Cilium-PR-Ginkgo-Tests-Validated/src/github.com/cilium/cilium/test/ginkgo-ext/scopes.go:341
19:29:14     Check Geneve mode [It]
19:29:14     /home/jenkins/workspace/Cilium-PR-Ginkgo-Tests-Validated/src/github.com/cilium/cilium/test/ginkgo-ext/scopes.go:373
19:29:14 
19:29:14     Expected
19:29:14         <bool>: false
19:29:14     to be true
19:29:14 
19:29:14     /home/jenkins/workspace/Cilium-PR-Ginkgo-Tests-Validated/src/github.com/cilium/cilium/test/k8sT/Tunnels.go:146
```

At least there were no problems building Envoy:
```
19:22:15     k8s1-1.7: [2,279 / 2,280] Linking envoy; 6s local
19:22:15     k8s1-1.7: 
19:22:24     k8s1-1.7: Target //:envoy up-to-date:
19:22:24     k8s1-1.7: 
19:22:24     k8s1-1.7:   bazel-bin/envoy
19:22:24     k8s1-1.7: 
19:22:24     k8s1-1.7: INFO: Elapsed time: 103.037s, Critical Path: 33.21s
```

**Comment by rlenglet** [comment]
test-me-please

**Comment by ianvernon** [comment]
Hit #4431: https://jenkins.cilium.io/job/Cilium-PR-Ginkgo-Tests-Validated/4058/

**Comment by ianvernon** [comment]
test-me-please

**Comment by eloycoto** [comment]
test-me-please

**Comment by eloycoto** [comment]
test-me-please

**Comment by eloycoto** [comment]
test-me-please

**Comment by eloycoto** [comment]
https://jenkins.cilium.io/job/Cilium-PR-Ginkgo-Tests-Validated/4080/testReport/k8s-1/7/K8sValidatedKafkaPolicyTest_Kafka_Policy_Tests_KafkaPolicies/

**Comment by eloycoto** [comment]
This is still WIP: 

```
===================== TEST FAILED =====================
cmd: kubectl get pods -o wide --all-namespaces
NAMESPACE     NAME                           READY     STATUS    RESTARTS   AGE       IP              NODE
default       deathstar-765fd545f9-4xtrf     1/1       Running   0          52s       10.10.1.78      k8s2
default       deathstar-765fd545f9-6sck5     1/1       Running   0          52s       10.10.1.169     k8s2
default       deathstar-765fd545f9-9cnl2     1/1       Running   0          52s       10.10.1.106     k8s2
default       spaceship-8595b78f8c-7tv4w     1/1       Running   0          52s       10.10.1.189     k8s2
default       spaceship-8595b78f8c-c79px     1/1       Running   0          52s       10.10.1.132     k8s2
default       spaceship-8595b78f8c-w4bmj     1/1       Running   0          52s       10.10.1.50      k8s2
default       spaceship-8595b78f8c-wzbkj     1/1       Running   0          52s       10.10.1.27      k8s2
default       xwing-77cb49f96-4rgn2          1/1       Running   0          28s       10.10.1.248     k8s2
default       xwing-77cb49f96-kqzkj          1/1       Running   0          28s       10.10.1.67      k8s2
default       xwing-77cb49f96-vrmkx          1/1       Running   0          28s       10.10.1.25      k8s2
kube-system   cilium-ccm6b                   1/1       Running   0          1m        192.168.36.12   k8s2
kube-system   cilium-qcvpr                   1/1       Running   0          1m        192.168.36.11   k8s1
kube-system   etcd-k8s1                      1/1       Running   0          8m        192.168.36.11   k8s1
kube-system   kube-apiserver-k8s1            1/1       Running   0          8m        192.168.36.11   k8s1
kube-system   kube-controller-manager-k8s1   1/1       Running   0          9m        192.168.36.11   k8s1
kube-system   kube-dns-59d8c5f9b5-p9rfn      3/3       Running   0          8m        10.88.0.2       k8s1
kube-system   kube-proxy-68gn6               1/1       Running   0          1m        192.168.36.12   k8s2
kube-system   kube-proxy-9wggj               1/1       Running   0          8m        192.168.36.11   k8s1
kube-system   kube-scheduler-k8s1            1/1       Running   0          9m        192.168.36.11   k8s1
```

DNS IPs are out of the range, `10.88.0.2` that are not controlled by Cilium at all. Some weird issue with the vagrant box for sure. 

This is related with CRIO: 
```
vagrant@k8s1:~$ cat /etc/cni/net.d/100-crio-bridge.conf 
{
    "cniVersion": "0.3.0",
    "name": "crio-bridge",
    "type": "bridge",
    "bridge": "cni0",
    "isGateway": true,
    "ipMasq": true,
    "ipam": {
        "type": "host-local",
        "subnet": "10.88.0.0/16",
        "routes": [
            { "dst": "0.0.0.0/0" }
        ]
    }
}
```
Regards

**Comment by eloycoto** [comment]
test-me-please

**Comment by eloycoto** [comment]
@aanm is the crio CNI needed at all? Or we can delete on packer-ci-build?

**Comment by ianvernon** [comment]
IMO, we should revert addition of cri-o into the VM, and push that as a new version to Vagrant Cloud. The addition of cri-o support can be re-added once it is tested locally, as it seems to not be trivial. 

Ideally we should have a build that tests against cri-o, which will require additional infrastructure on the test framework side (VM build options, etc.), as well as tests which test Cilium integration. This should be the bar for which addition of cri-o into the VM should be measured so that addition of other, unrelated functionality into the VM is not blocked.

**Comment by ianvernon** [comment]
I reverted the addition of cri-o into the VM build here: https://github.com/cilium/packer-ci-build/pull/66

**Comment by aanm** [comment]
@eloycoto `/etc/cni/net.d/100-crio-bridge.conf ` can be removed as a workaround

**Comment by eloycoto** [comment]
test-me-please

**Comment by eloycoto** [comment]
test-me-please

**Comment by eloycoto** [comment]
Few Jenkins that failed here, and I can replicated quite easy, if kube-dns is in place **before** cilium, the system will not start correctly on Kubernetes 1.7.

**Comment by eloycoto** [comment]
test-me-please

**Comment by tgraf** [comment]
> Few Jenkins that failed here, and I can replicated quite easy, if kube-dns is in place before cilium, the system will not start correctly on Kubernetes 1.7.

How about we delete the kube-dns pods after deploying Cilium for the first time to make it reliable?

**Comment by eloycoto** [comment]
@tgraf fixed, was some loopback CNI with CRI-O in. Should be ok ;-)

**Comment by eloycoto** [comment]
test-me-please

**Review by ianvernon — APPROVED** [review]

**Comment by ianvernon** [comment]
Given that @aanm is janitor this week, this already has janitor approval. Thus, I am merging.

**Pull request merged**